luarules/gadgets/unit_no_attack_capture_target.lua

```diff
@@ -0,0 +1,115 @@
+function gadget:GetInfo()
+	return {
+		name		= "Don't attack capture target",
+		desc		= "Stops your units from targeting your capture targets",
+		author		= "Slouse",
+		version		= 'v1.0',
+		date		= "April 2025",
+		license		= "GNU GPL, v2 or later",
+		layer		= -1,
+		enabled 	= true
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then
+    return false
+end
+
+local captureTargets = {}
+
+local function activeCaptureCommand(unitID)
+	local cmdQueue = Spring.GetUnitCommands(unitID, -1);
+	if #cmdQueue > 0 then
+		if cmdQueue[1].id == CMD.CAPTURE then
+			return true
+		else
+			return false
+		end
+	end
+end
+
+--ToDo: Determine which units weapons need to be watched
+--
+--Watch all weapons so gadget:AllowWeaponTarget() works
+for unitDefID, unitDef in pairs(UnitDefs) do
+	local weapons = unitDef.weapons
+	for i = 1, #weapons do
+		for wid, weapon in ipairs(unitDef.weapons) do
+			if weapon ~= nil then
+				local weaponDef = weapon.weaponDef
+				if weaponDef ~= nil then
+					Script.SetWatchWeapon(weaponDef, true)
+				end
+			end
+		end
+	end
+end
+
+--Mark target for capture when capture command is accepted by unit
+function gadget:UnitCommand(unitID, unitDefID, teamID, cmdID, cmdParams, cmdOptions, cmdTag, playerID, fromSynced, fromLua)
+	if cmdID == CMD.CAPTURE then
+		local targetID = cmdParams[1]
+		--If targetID is already in captureTargets table, ignore it
+		if not captureTargets[targetID] then
+			captureTargets[targetID] = {builderID = unitID, captureTeamID = teamID, captureProgress = 0}
+		end
+	end
+end
+
+--Remove target from captureTargets table when capture is completed
+function gadget:AllowUnitTransfer(unitID, unitDefID, oldTeam, newTeam, capture)
+	if capture == true then
+		if captureTargets ~= nil then
+			if captureTargets[unitID] then
+				captureTargets[unitID] = nil
+			end
+		end
+	end
+	return true
+end
+
+--Remove unit from captureTargets table if destroyed
+function gadget:UnitDestroyed(unitID, unitDefID, unitTeam, attackerID, attackerDefID, attackerTeam, weaponDefID)
+	if captureTargets ~= nil then
+		if captureTargets[unitID] then
+			captureTargets[unitID] = nil
+		end
+	end
```

Comment:
Can the `captureTargets` table ever be nil? You don't check it in line 53 and it's not getting set to nil anywhere so I assume not. Perhaps just
```suggestion
	captureTargets[unitID] = nil
```